unit/unittype.go

```diff
@@ -130,6 +130,31 @@ var (
 // dimension is zero. Dimensions is used in conjunction with New.
 type Dimensions map[Dimension]int
 
+func (d Dimensions) clone() Dimensions {
+	if len(d) == 0 {
+		return nil
+	}
+	c := make(Dimensions, len(d))
+	for dim, pow := range d {
+		if pow != 0 {
```

Comment:
(wrt below) this shouldn't be necessary because all of the 0 dimensions shouldn't have keys. On the one hand, it's not bad to check, on the other hand if we have some failure of this invariant we disguise it. Not sure what the best plan is.

Reply:
It is necessary to have it here, because `Dimensions` is an exported and user-mutable type. We must remove any zeros prior to use. The zeroing on handing back to the user is a side-effect.